### PR TITLE
Tweak the API probes to accommodate xray and newrelic init

### PR DESCRIPTION
**Pull request by jimleroyer**
## What happens when your PR merges?

The API probes should be more reliable for the API pods deployment. 

## What are you changing?

- [ ] Releasing a new version of Notify
- [ ] Changing kubernetes configuration

## Provide some background on the changes

With the recent upgrade of New Relic to version 9.2.0, we reviewed the existing probes (startup, liveness and readiness) for the API, to take in account initialization of the former and of AWS X-Ray. As the startup time of our application can get longer, we need to accommodate for these. 

## If you are releasing a new version of Notify, what components are you updating

- [ ] API
- [ ] Admin
- [ ] Documentation
- [ ] Document download API

## Checklist if releasing new version

- [ ] I made sure that the changes are as expected in [Notify staging](https://staging.notification.cdssandbox.xyz/)
- [ ] I have checked if the docker images I am referencing exist
  - [ ] [api lambda](https://ca-central-1.console.aws.amazon.com/ecr/repositories/private/296255494825/notify/api-lambda?region=ca-central-1) (requires Notification-Production / AdministratorAccess login)
  - [ ] [api k8s](https://gallery.ecr.aws/v6b8u5o6/notify-api)
  - [ ] [admin](https://gallery.ecr.aws/v6b8u5o6/notify-admin)
  - [ ] [documentation](https://gallery.ecr.aws/v6b8u5o6/notify-documentation)
  - [ ] [document download API](https://gallery.ecr.aws/v6b8u5o6/notify-document-download-api)

## Checklist if making changes to Kubernetes

- [ ] I know how to get kubectl credentials in case it catches on fire

## After merging this PR

- [ ] I have verified that the tests / deployment actions succeeded
- [ ] I have verified that any affected pods were restarted successfully
- [ ] I have verified that I can still log into [Notify production](https://notification.canada.ca)
- [ ] I have verified that the smoke tests still pass on production
- [ ] I have communicated the release in the #notify Slack channel.

### CODE REVIEW

**Review by P0NDER0SA — APPROVED** [review]
This is good!

File: base/notify-api/api-deployment.yaml [hunk 1]
```diff
@@ -287,11 +287,11 @@ spec:
             httpGet:
               path: /_status?simple=true
               port: 6011
-            initialDelaySeconds: 15
+            initialDelaySeconds: 20
             periodSeconds: 3
             timeoutSeconds: 1
             successThreshold: 1
-            failureThreshold: 6
+            failureThreshold: 20
```
Comment (jimleroyer) [inline base/notify-api/api-deployment.yaml:294]: We are total giving 1m20s of start up time to the app, with 20 seconds being the minimum initialization and then trying to get the ball rolling with a http test every 3 seconds, for 1 minute.

File: base/notify-api/api-deployment.yaml [hunk 2]
```diff
@@ -300,12 +300,12 @@ spec:
             periodSeconds: 3
             timeoutSeconds: 1
             successThreshold: 3
-            failureThreshold: 5
+            failureThreshold: 10
```
Comment (jimleroyer) [inline base/notify-api/api-deployment.yaml:303]: Once the startup probe is done, the readiness and liveness probes will kick in. The readiness will determine when the app is ready and might need to wait more time than what the startup probe allowed. The reason for that is that the first http hit will trigger an initialization for lazy loaded components such as New Relic. So we need to give it more flexibility. 

A failure threshold of 10 for 3 period seconds means that a total of 30 seconds will be tried, but after waiting for an initial delay seconds of 5 seconds. Hence 35 seconds is allowed for this readiness stage.

File: base/notify-api/api-deployment.yaml (hunk 1 above)
Comment (P0NDER0SA) [reply base/notify-api/api-deployment.yaml:294]: Interesting!

**Pull request merged**